app/assets/stylesheets/govuk_publishing_components/components/_search.scss

```diff
@@ -1,6 +1,21 @@
 $input-size: 40px;
 $large-input-size: 50px;
 
+%search-input-focus {
```

Comment:
I'd be tempted to avoid the placeholder class and just extend the original selector (focus n' all) – even though it means disabling another linting rule.

```scss
.gem-c-search__input[type="search"] {

  // ...

  &:focus {
    outline: $govuk-focus-width solid $govuk-focus-colour;
    // Ensure outline appears outside of the element
    outline-offset: 0;
    // Double the border by adding its width again. Use `box-shadow` for this // instead of changing `border-width`
    // Also, `outline` cannot be utilised here as it is already used for the yellow focus state.
    box-shadow: inset 0 0 0 $govuk-border-width-form-element;

    @include govuk-if-ie8 {
      // IE8 doesn't support `box-shadow` so double the border with
      // `border-width`.
      border-width: $govuk-border-width-form-element * 2;
    }
  }
}

// Override overly specific focus styles from govuk_template
@include govuk-compatibility(govuk_template) {
  #global-header .gem-c-search__input[type="search"]:focus { // stylelint-disable selector-max-id
    @extend .gem-c-search__input[type="search"]:focus; // stylelint-disable scss/at-extend-no-missing-placeholder
  }
}
```

That way the 'nice' code is untouched, and the 'hacky' override for template is all neatly contained in one place – and when (one day, when template is retired) we come to clean up the code we just need to remove the block wrapped in `govuk-compatibility` .

Reply:
Thank you for this comment @36degrees!
Good point there: we want to contain this neatly to make it easier for future us. 

I've updated this based on your suggestion. 👍